### PR TITLE
Rename resolve-handles to resolve-recipe, and start resolving slots.

### PR DESCRIPTION
**Pull request by shans**
This follows the pattern already demonstrated in resolve-recipe -
only resolve if there's one and only one option.

Also modify map-slots to only resolve if there are choices.

### CODE REVIEW

File: runtime/strategies/map-slots.js [hunk 1]
```diff
@@ -72,12 +70,10 @@ export default class MapSlots extends Strategy {
 
       // Helper methods.
       // Chooses the best slot out of the given slot candidates.
```
Comment (dstockwell) [inline runtime/strategies/map-slots.js]: Doesn't do this anymore.

File: runtime/strategies/resolve-recipe.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2017 Google Inc. All rights reserved.
```
Comment (dstockwell) [inline runtime/strategies/resolve-recipe.js]: 2018

File: runtime/strategies/resolve-recipe.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            `Cannot override slot id '${slotConnection.targetSlot.id}' with '${selectedSlot.id}'`);
+          slotConnection.targetSlot.id = selectedSlot.id || slotConnection.targetSlot.id;
+
+          // TODO: need to concat to existing tags and dedup?
+          slotConnection.targetSlot.tags = [...selectedSlot.tags];
+          return 1;
+        };
+      }
+
+      _slotIsConnected(slotConnection) {
```
Comment (dstockwell) [inline runtime/strategies/resolve-recipe.js]: todo move to SlotConnection

File: runtime/strategies/resolve-recipe.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        return slotConnection.targetSlot && ((!!slotConnection.targetSlot.sourceConnection) || (!!slotConnection.targetSlot.id));
+      }
+
+      _findSlotCandidates(slotConnection, slots) {
```
Comment (dstockwell) [inline runtime/strategies/resolve-recipe.js]: For now, I think we could make these functions on MapSlots be `static` (I don't think they use `this` other than to call each other)? I think much of this should ultimately be moved to recipe/

File: runtime/strategies/map-slots.js (hunk 1 above)
Comment (shans) [reply runtime/strategies/map-slots.js]: It never did!!!

File: runtime/strategies/resolve-recipe.js (hunk 3 above)
Comment (shans) [reply runtime/strategies/resolve-recipe.js]: Done.

File: runtime/strategies/resolve-recipe.js (hunk 4 above)
Comment (shans) [reply runtime/strategies/resolve-recipe.js]: Done

File: runtime/strategies/resolve-recipe.js (hunk 2 above)
Comment (shans) [reply runtime/strategies/resolve-recipe.js]: Done

**Comment by mmandlis** [comment]
Shane, could you update https://github.com/PolymerLabs/arcs/blob/master/runtime/strategies.md with a brief description of the new strategies? thanks.

**Pull request merged**